### PR TITLE
Not make fsspec hard dependency

### PR DESCRIPTION
**Pull request by martindurant**
...but it is needed in bag, array and dataframe

- [ ] Tests added / passed
- [x] Passes `black dask` / `flake8 dask`

### CODE REVIEW

File: dask/dataframe/io/hdf.py [hunk 1]
```diff
@@ -8,6 +8,7 @@
 
 import pandas as pd
 from toolz import merge
+from ...bytes import read_bytes  # noqa
```
Comment (TomAugspurger) [inline dask/dataframe/io/hdf.py]: May want a comment here explaining that importing `dask.bytes` has a side-effect of checking for fsspec.

**Comment by martindurant** [comment]
Just partd again

**Comment by shoyer** [comment]
Do we really need fsspec for dask.array? Not that I’m strongly opposed, but it’s pretty common not to use any of dask’s builtin IO with dask.array. The most popular storage formats are all external, likely Zarr, HDF5 and netCDF.

**Comment by martindurant** [comment]
dask.array does indeed use fsspec, but only for zarr (to create the mappers across whichever backend). Since you can also specify the mapper directly, I've made it optional.

**Pull request merged**